### PR TITLE
Persistence Component: Fix for empty table check SQL for snowflake

### PR DESCRIPTION
**Pull request by prasar-ashutosh**
#### What type of PR is this?
Bug Fix

#### What does this PR do / why is it needed ?
Bug fix for SQL that checks if the table is empty.

#### Which issue(s) this PR fixes:
Bug fix for SQL that checks if the table is empty.
The old sql was not compliant with Snowflake
Rewrote the SQL that is now compliant with Snowflake and added related tests.
#### Does this PR introduce a user-facing change?
NO

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/LogicalPlanFactory.java [hunk 1]
```diff
@@ -41,14 +40,14 @@
...
     public static final String IS_TABLE_NON_EMPTY = "isTableNonEmpty";
     public static final String TABLE_IS_NON_EMPTY = "1";
+    public static final String TABLE_ALIAS = "X";
 
     public static LogicalPlan getLogicalPlanForIsDatasetEmpty(Dataset dataset)
     {
-        Condition exists = Exists.of(Selection.builder().source(dataset).addAllFields(LogicalPlanUtils.ALL_COLUMNS()).build());
-        Function count = FunctionImpl.builder().functionName(FunctionName.COUNT).addValue(NumericalValue.of(1L)).alias(IS_TABLE_NON_EMPTY).build();
-
+        Function count = FunctionImpl.builder().functionName(FunctionName.COUNT).addValue(All.INSTANCE).alias(IS_TABLE_NON_EMPTY).build();
```
Comment (rengam32) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/LogicalPlanFactory.java:47]: Can we rename to isTableNonEmpty or so?

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/sqldom/schemaops/statements/SelectStatement.java [hunk 2]
```diff
@@ -128,6 +129,15 @@ public void genSql(StringBuilder builder) throws SqlDomException
             }
         }
 
+        // Add limit clause
```
Comment (rengam32) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/sqldom/schemaops/statements/SelectStatement.java:132]: Can we change the comment on line 76 to limit

Comment (prasar-ashutosh) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/sqldom/schemaops/statements/SelectStatement.java:132]: Will add that

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/LogicalPlanFactory.java (hunk 1 above)
Comment (prasar-ashutosh) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/LogicalPlanFactory.java:47]: Sure, will rename it to countAllRows

**Review by kumuwu — APPROVED** [review]

**Review by rafaelbey — APPROVED** [review]

**Pull request merged**